src/main/kotlin/app/domain/DomainClasses.kt

```diff
@@ -7,6 +7,29 @@ data class EncryptionMetadata(val initializationVector: String,
                               val cipherText: String,
                               var plaintext: String)
 
-data class EncryptedStream(var inputStream: InputStream, val fileName: String, val fullPath: String, val encryptionMetadata: EncryptionMetadata)
+data class EncryptedStream(var contents: ByteArray, val fileName: String, val fullPath: String, val encryptionMetadata: EncryptionMetadata) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as EncryptedStream
+
+        if (!contents.contentEquals(other.contents)) return false
+        if (fileName != other.fileName) return false
+        if (fullPath != other.fullPath) return false
+        if (encryptionMetadata != other.encryptionMetadata) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = contents.contentHashCode()
+        result = 31 * result + fileName.hashCode()
```

Comment:
why 31?

Reply:
It's autogenerated - I think it just means the hashes are evenly spread.
https://stackoverflow.com/questions/299304/why-does-javas-hashcode-in-string-use-31-as-a-multiplier